admin/client/EditorCustomizeTab.tsx

```diff
@@ -37,30 +49,104 @@ class ColorSchemeSelector extends React.Component<{ chart: ChartConfig }> {
+                    alignItems: "center"
+                }}
+            >
+                <div>{option.label}</div>
+                <span
+                    style={{
+                        backgroundImage: this.createLinearGradient(
+                            option.colorScheme,
+                            6
+                        ),
+                        width: "6rem",
+                        height: "1.25rem",
+                        border: "1px solid #aaa"
+                    }}
+                />
+            </div>
+        )
+    }
+
     render() {
         const { chart } = this.props
 
-        const availableColorSchemes = keysOf(ColorSchemes)
-        const colorSchemeLabels = availableColorSchemes.map(
-            scheme => (ColorSchemes[scheme] as ColorScheme).name
+        const defaultColorScheme = {
+            value: "default",
+            label: "Default",
+
+            // This is the default color in DiscreteBarTransform
+            colorScheme: new ColorScheme(
+                "DiscreteBarChart default",
+                [["#2e5778", "#2e5778", "#2e5778"]],
+                false
+            )
+        }
+
+        const availableColorSchemes = [defaultColorScheme].concat(
+            Object.entries(ColorSchemes).map(([key, scheme]) => {
+                return {
+                    value: key,
+                    label: (scheme as ColorScheme).name,
+                    colorScheme: scheme as ColorScheme
+                }
+            })
         )
 
         return (
-            <FieldsRow>
-                <SelectField
-                    label="Color scheme"
-                    value={chart.baseColorScheme || "default"}
-                    onValue={this.onValue}
-                    options={["default"].concat(availableColorSchemes)}
-                    optionLabels={["Default"].concat(colorSchemeLabels)}
-                />
-                <br />
-                <Toggle
-                    label="Invert colors"
-                    value={!!chart.props.invertColorScheme}
-                    onValue={this.onInvertColorScheme}
-                />
-            </FieldsRow>
+            <React.Fragment>
+                <FieldsRow>
+                    <Select
+                        defaultValue={
+                            availableColorSchemes.find(
+                                scheme => scheme.value === chart.baseColorScheme
+                            ) || defaultColorScheme
+                        }
+                        label="Color scheme"
+                        options={availableColorSchemes}
+                        onChange={this.onChange}
+                        formatOptionLabel={this.formatOptionLabel}
+                        styles={{
+                            singleValue: provided => {
+                                return {
+                                    ...provided,
+                                    width: "calc(100% - 10px)"
```

Comment:
I'm not quite happy with this hardcoded width in here (SingleValue has a `max-width: calc(100% - 8px)` as well), but couldn't find another way to get the flex element to spread.

Reply:
Ah yes, I couldn't figure this out either. But I think this is good enough, even if the values change in the future and introduce some clipping... it's alright :)